### DISCUSSION TITLE
Difference between KC_HTTP_MAX_QUEUED_REQUESTS and QUARKUS_THREAD_POOL_QUEUE_SIZE

### DISCUSSION THREAD

**Discussion by durgasusm** [keycloak/keycloak#26244] [Q&A]
Hi,

What should be the ideal value of KC_HTTP_MAX_QUEUED_REQUESTS since it has been introduced in https://www.keycloak.org/server/all-config#category-http
What is the Difference between KC_HTTP_MAX_QUEUED_REQUESTS and QUARKUS_THREAD_POOL_QUEUE_SIZE?
If both are defined which value should be taken as the priority ?

**Comment by vmuzikar** [comment]
> What should be the ideal value of KC_HTTP_MAX_QUEUED_REQUESTS

Please see https://www.keycloak.org/high-availability/concepts-threads#load-shedding. It still mentions `quarkus.thread-pool.max-threads` but will be [updated](https://github.com/keycloak/keycloak/commit/ba76682590a51bade98ccb90e6d6d9087165cdef) to `http-max-queued-requests` in the next major release.

> What is the Difference between KC_HTTP_MAX_QUEUED_REQUESTS and QUARKUS_THREAD_POOL_QUEUE_SIZE?

They do the same. The Keycloak option just sets the Quarkus option. Keycloak option is considered supported while setting raw Quarkus options is not supported and serves just as an escape hatch.
